sdk/core/Azure.Core/tests/ModelSerializationTests/Animal.cs

```diff
@@ -159,6 +159,43 @@ public bool TrySerialize(Stream stream, out long bytesWritten, SerializableOptio
                 return false;
             }
         }
+
+        public void Deserialize(Stream stream, SerializableOptions options = default)
+        {
+            try
+            {
+                JsonDocument jsonDocument = JsonDocument.Parse(stream);
+                var model = DeserializeAnimal(jsonDocument.RootElement, options ?? new SerializableOptions());
+                this.LatinName = model.LatinName;
+                this.Weight = model.Weight;
+                this.IsHungry = model.IsHungry;
+                this.Name = model.Name;
+                this.RawData = model.RawData;
+            }
+            catch
+            {
+                throw new RequestFailedException("Model could not be deserialized.");
```

Comment:
no need to catch and throw our own exception here just let the exceptions bubble up